src/basic/cgroup-util.c

```diff
@@ -140,6 +142,12 @@ int cg_read_pidref(FILE *f, PidRef *ret) {
                         return 0;
                 }
 
+                if (pid == 0)
+                        /* An unmappable PID. We certainly cannot create a pidref for it, so ignore
+                         * it like other PIDs that we cannot find. (Also 0 would be interpreted as
+                         * us by pidref_set_pid(), which we cannot allow to happen.) */
+                        continue;
```

Comment:
Hmm, now `cg_read_pidref` skips over `0` pids. This implication seems somehow weirder compared to the flags approach.

E.g. I think `unit_search_main_pid` should refuse 0 pids, since for services the cgroup should really be fully controlled by us, and unmappable pids are inherently an error.

So ideally, when the flag is set for `cg_read_pid`, it should return `0` pid. But when set for `cg_read_pidref`, it should return an error. To make this controllable, I think flags are needed.